File: backend/src/main/java/com/example/exammanagement/service/StudentService.java
package com.example.exammanagement.service;

import com.example.exammanagement.entity.Student;
import com.example.exammanagement.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StudentService {
    
    @Autowired
    private StudentRepository studentRepository;
    
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }
    
    public Student getStudentById(Long id) {
        return studentRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Student not found with id: " + id));
    }
    
    public List<Student> getStudentsByCourse(Long courseId) {
        return studentRepository.findByCourseId(courseId);
    }
    
    public Student createStudent(Student student) {
        return studentRepository.save(student);
    }
    
    public Student updateStudent(Long id, Student studentDetails) {
        Student student = getStudentById(id);
        
        student.setName(studentDetails.getName());
        student.setEmail(studentDetails.getEmail());
        student.setRollNumber(studentDetails.getRollNumber());
        student.setPhoneNumber(studentDetails.getPhoneNumber());
        student.setCourses(studentDetails.getCourses());
        
        return studentRepository.save(student);
    }
    
    public void deleteStudent(Long id) {
        Student student = getStudentById(id);
        studentRepository.delete(student);
    }
}

File: frontend/src/services/studentService.js
import api from './api'

export const studentService = {
  async getStudents() {
    const response = await api.get('/students')
    return response.data
  },

  async getStudentById(id) {
    const response = await api.get(`/students/${id}`)
    return response.data
  },

  async createStudent(studentData) {
    const response = await api.post('/students', studentData)
    return response.data
  },

  async updateStudent(id, studentData) {
    const response = await api.put(`/students/${id}`, studentData)
    return response.data
  },

  async deleteStudent(id) {
    const response = await api.delete(`/students/${id}`)
    return response.data
  },

  async getStudentsByCourse(courseId) {
    const response = await api.get(`/students/course/${courseId}`)
    return response.data
  },
}
